src/Sylius/Bundle/OrderBundle/NumberGenerator/OrderNumberGeneratorInterface.php

```diff
@@ -11,13 +11,17 @@
 
 namespace Sylius\Bundle\OrderBundle\NumberGenerator;
 
+use Sylius\Component\Order\Model\OrderInterface;
+
 /**
  * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
  */
-interface SequentialOrderNumberGeneratorInterface
+interface OrderNumberGeneratorInterface
 {
     /**
+     * @param OrderInterface $order
+     *
      * @return string
      */
-    public function generate();
+    public function generate(OrderInterface $order);
```

Comment:
You may want to generate order number based on: channel code, date, customer group... whatever. This gives us access to all info about the order. Otherwise we have 0.